Resource.js

```diff
@@ -1,8 +1,15 @@
 var _ = require('lodash');
 var mongoose = require('mongoose');
 var paginate = require('node-paginate-anything');
+var patcher = require('mongoose-json-patch');
 
-module.exports = function(app, route, modelName, model) {
+module.exports = function(app, route, modelName, schema) {
+
+  // Attach the mongoose-json-patch plugin
+  schema.plugin(patcher);
+
+  // Generate the mongoose model
+  var model= mongoose.model(modelName, schema);
```

Comment:
I have an opinion about this... 

The reason why I was keeping the model declaration out of this library is because it is common for someone to declare the model somewhere else in their library where it is used in other placed.  Here we would be creating a separate instance of that model which make consume more memory as well as cause unwanted consequences...

Were you having problems passing the ```mongoose.model``` class to this library?